mp/src/game/shared/momentum/util/engine_patch.cpp

```diff
@@ -0,0 +1,243 @@
+    //    (unsigned char*)"\xC1\xEF\x03\x81\xFF\x00\x00\x00\x00",
+    //    "xxxxx????",
+    //    5,
+    //    true,
+    //    (unsigned char*)"\x00\x40\x00\x00"
+    //}
+};
+#elif __linux__
+CEnginePatch g_EnginePatches[] =
+{
+    // Prevent the culling of skyboxes at high FOVs
+    // https://github.com/VSES/SourceEngine2007/blob/master/se2007/engine/gl_warp.cpp#L315
+    // TODO: Use a value derived from FOV instead
+    {
+        "SkyboxCulling",
+        (unsigned char*)"\xF3\x0F\x59\x0D\x00\x00\x00\x00\xF3\x0F\x58\xC2\xF3\x0F\x58\xC1\xF3\x0F\x10\x0D",
+        "xxxx????xxxxxxxxxxxx",
+        20,
+        false,
+        (float)-1
+    }
+};
+#endif //WIN32
+
+void CEnginePatch::ApplyAll()
+{
+#if !defined (OSX) // No OSX patches yet
+    int err;
+
+    for (int i = 0; i < sizeof(g_EnginePatches) / sizeof(*g_EnginePatches); i++)
+    {
+        err = g_EnginePatches[i].ApplyPatch();
+        switch (err)
+        {
+        case 1:
+            Warning("Engine patch \"%s\" FAILED: Could not override memory protection\n", g_EnginePatches[i].getName());
+            break;
+        case 2:
+            Warning("Engine patch \"%s\" FAILED: Could not find signature\n", g_EnginePatches[i].getName());
+        }
+    }
+#endif // (OSX)
+}
+
+// Apply a patch to engine memory
+int CEnginePatch::ApplyPatch()
+{
+#ifdef _WIN32
+
+    uintptr_t addr = reinterpret_cast<uintptr_t>(CEngineBinary::FindPattern(m_pPatchSignature, m_pMask, m_iOffset));
+
+    if (addr)
+    {
+        auto pMemory = m_bImmediate ? (uintptr_t*)addr : *reinterpret_cast<uintptr_t**>(addr);
+
+        // Memory is write-protected so it needs to be lifted before the patch is applied
+        // 0x40 is read,write,execute
+        // https://docs.microsoft.com/en-us/windows/win32/memory/memory-protection-constants
+        unsigned long iOldProtection, iNewProtection = 0x40;
+
+        if (VirtualProtect(pMemory, m_iPatchLength, iNewProtection, &iOldProtection))
+        {
+            Q_memcpy(pMemory, m_pPatch, m_iPatchLength);
+
+            // Restore old protections
+            VirtualProtect(pMemory, m_iPatchLength, iOldProtection, &iNewProtection);
+
+            return 0;
+        }
+        return 1;
+    }
+    return 2;
+
+#elif defined __linux__
+
+    void* addr = CEngineBinary::FindPattern(m_pPatchSignature, m_pMask, m_iOffset);
+
+    if (addr)
+    {
+        auto pMemory = m_bImmediate ? (uintptr_t*)addr : *(uintptr_t**)(addr);
+
+        // Memory is write-protected so it needs to be lifted before the patch is applied
+        // http://man7.org/linux/man-pages/man2/mprotect.2.html
+        if (!mprotect(pMemory, m_iPatchLength, PROT_READ | PROT_WRITE | PROT_EXEC))
+        {
+            Q_memcpy(pMemory, m_pPatch, m_iPatchLength);
+
+            // Restore old protections
+            mprotect(pMemory, m_iPatchLength, PROT_READ | PROT_EXEC);
+
+            return 0;
+        }
+        return 1;
+    }
+    return 2;
+
+#endif //WIN32
```

Comment:
The 2 blocks are quite similar, the only difference between the 2 is the OS specific memory protection changes. You should move that memory protection code specifically into it's own function/class and merge the 2 blocks.